tools/mo/openvino/tools/mo/back/preprocessing.py

```diff
@@ -399,6 +393,16 @@ def apply_preprocessing(ov_function: Model, argv: argparse.Namespace):
             else:
                 prep.output(node_name).tensor().set_layout(Layout(layout_value['target_layout']))
 
+    # Apply pre-processing builder to a function
+    ov_function = prep.build()
+
+    need_reverse = 'reverse_input_channels' in argv and argv.reverse_input_channels
+    suitable_params_ric = []
+    if need_reverse:
+        suitable_params_ric = guess_source_layouts_for_reverse_channels(ov_function=ov_function,
```

Comment:
Is it possible to fix it in 'guess_source_layout_for_reverse_channels'?
https://github.com/mvafin/openvino/blob/85e40990821739463b571954e6c3849f6fa00850/tools/mo/openvino/tools/mo/back/preprocessing.py#L265
This function shall detect 'C' channel using 'layout_values' information even if layout is not yet set to Parameter
`Additionally checks existing layouts and detects suitable inputs for reversing of input channels`

Reply:
Yes, but there is a problem of how to look for names. We need to look through all tensor names and friendly_name, I think that overhead of running preprocessing twice shouldn't be big, what do you think would be the best?